service_client.go

```diff
@@ -85,19 +82,14 @@ func NewInMemoryServiceClient() (ServiceClient, error) {
 func NewPersistentServiceClient(path string) (ServiceClient, error) {
 	var opts Opts
 
-	bopts := badger.DefaultOptions
-	bopts.ValueLogLoadingMode = options.FileIO
-
-	ds, err := badger.NewDatastore(path, &bopts)
-	if err != nil {
-		return nil, fmt.Errorf("unable to init badger datastore: %w", err)
-	}
-
 	repo, err := ipfsutil.LoadRepoFromPath(path)
 	if err != nil {
 		return nil, err
 	}
 
+	// use ipfs root datastore
+	opts.RootDatastore = repo.Datastore()
```

Comment:
you should probably create a namespaced datastore from the ipfs datastore

Reply:
you totally right, thanks !